panel/tests/ui/test_auth.py

```diff
@@ -39,6 +39,100 @@ def test_basic_auth(py_file, page, prefix):
         expect(page.locator('.markdown')).to_have_text('test_user', timeout=10000)
 
 
+@unix_only
+@pytest.mark.skipif('OKTA_OAUTH_KEY' not in os.environ, reason='Okta credentials not available')
+def test_okta_oauth(py_file, page):
+    app = "import panel as pn; pn.pane.Markdown(pn.state.user).servable(title='A')"
+    write_file(app, py_file.file)
+
+    port = os.environ.get('OKTA_PORT', '5703')
+    cookie_secret = os.environ['OAUTH_COOKIE_SECRET']
+    encryption_key = os.environ['OAUTH_ENCRYPTION_KEY']
+    oauth_key = os.environ['OKTA_OAUTH_KEY']
+    oauth_secret = os.environ['OKTA_OAUTH_SECRET']
+    extra_params = os.environ['OKTA_OAUTH_EXTRA_PARAMS']
+    okta_user = os.environ['OKTA_OAUTH_USER']
+    okta_password = os.environ['OKTA_OAUTH_PASSWORD']
+    cmd = [
+        "--port", port, "--oauth-provider", "okta", "--oauth-key", oauth_key,
+        "--oauth-secret", oauth_secret, "--cookie-secret", cookie_secret,
+        "--oauth-encryption-key", encryption_key, "--oauth-extra-params", extra_params,
+        py_file.name
+    ]
+    with run_panel_serve(cmd) as p:
+        port = wait_for_port(p.stdout)
+        page.goto(f"http://localhost:{port}")
+
+        page.locator('input[name="username"]').fill(okta_user)
+        page.locator('input[name="password"]').fill(okta_password)
+        page.locator('input[type="submit"]').click(force=True)
+
+        expect(page.locator('.markdown')).to_have_text(okta_user, timeout=10000)
+
+
+@unix_only
+@pytest.mark.skipif('AZURE_OAUTH_KEY' not in os.environ, reason='Azure credentials not available')
```

Comment:
These checks seem common enough for people to actually have in their environment. 

Maybe have an environment variable called `PANEL_CI_TESTS` which is used instead.

Reply:
I can prefix them with `PANEL_`

Reply:
That is also a way to do it :slightly_smiling_face:

Reply:
Honestly, I don't think it's very likely and what's the worst case? The test fails, that's honestly fine.